File: problems/queue_machine/submissions/accepted/queue_machine.py
def NEW(queue, ticket_number):
    queue.append(ticket_number)
    return queue


def CANCEL(queue, n_order):
    index = queue.index(n_order)
    queue.pop(index)
    return queue


def SHOW(queue, index):
    if index == "FIRST":
        print(queue[0])
    elif index == "LAST":
        print(queue[-1])
    else:
        print_queue(queue)


def NEXT(queue):
    if len(queue):
        queue.pop(0)
    return queue


def print_queue(queue):
    if len(queue):
        queue_to_string = ' '.join([str(elem) for elem in queue])
        print(queue_to_string)
    else:
        print(0)


if __name__ == "__main__":
    ticket_number = 0
    queue = []
    commands = []
    # getting the commands
    n_of_commands = int(input())
    for i in range(n_of_commands):
        commands.append(input().strip())
    # executing the commands
    for command in commands:
        if command == "NEW":
            ticket_number += 1
            queue = NEW(queue, ticket_number)
        elif ("CANCEL" in command):
            n_order = int(command.split(' ')[1])
            queue = CANCEL(queue, n_order)
        elif "SHOW" in command:
            index = command.split(' ')[1]
            SHOW(queue, index)
        else:
            queue = NEXT(queue)

File: problems/queue_machine/submissions/accepted/queue_machine.java
import java.util.ArrayList;
import java.util.Scanner;

public class QueueMachine {
    public static int ticketNumber = 0;
    public static ArrayList<String> queue = new ArrayList<String>();


    public static void addNew(){
        ticketNumber++;
        queue.add(String.valueOf(ticketNumber));
    }

    public static void cancel(int nbOrder){
        int index = queue.indexOf(String.valueOf(nbOrder));
        queue.remove(index);
    }

    public static void show(String index){
        switch (index) {
            case "FIRST":
                System.out.println(queue.get(0));
                break;
            case "LAST":
                System.out.println(queue.get(queue.size()-1));
                break;
            default:
                String q = queue.toString().replace(", ", " ").replaceAll("([\\[\\]])", "");
                System.out.println(q.length()>0?  q : "0");
        }
    }
    public static void next(){
        if(queue.size()>0){
            queue.remove(0);
        }
    }

    public static void main(String[] args) {
        ArrayList<String> commands = new ArrayList<String>();
        int nbCommands = 0;

        Scanner sc = new Scanner(System.in);
        nbCommands = Integer.parseInt(sc.nextLine());
        // getting the commands
        for(int i=0 ; i<nbCommands; i++){
            commands.add(sc.nextLine().strip());
        }   

        // executing commands 
        for (String command : commands) {

            if(command.equals("NEW")){
                addNew();
            }else{
                if (command.contains("CANCEL")) {
                    int nbOrder = Integer.parseInt(command.split(" ")[1]);
                    cancel(nbOrder);
                    
                } else {
                    if (command.contains("SHOW")) {
                        String index = command.split(" ")[1];
                        show(index);
                    } else{
                        next();
                    }
                }
            }
        }
        sc.close();
    }
}